src/views/AgmExtension.vue

```diff
@@ -274,7 +274,7 @@ export default class AgmExtension extends Mixins(CommonMixin, DateMixin, FilingM
 
     // this is the id of THIS filing
     // it must be 0 (meaning new filing) -- we do not support resuming a draft filing
-    this.filingId = +this.$route.params.filingId // number or NaN
+    this.filingId = +this.$route.query.filingId // number or NaN
```

Comment:
This is a check that all local filings make.

Since only an internal route can set the `params`, this ensured that users could not enter a filing route manually in order to file something they would normally have access to.

This is now changed to `query` (eg, "?filingId=123") so that the filings can be routed to from an external UI.

Any thoughts on check we should do to prevent users from manually entering a local route? Or should we not worry about it because the Legal API will check if the filing is allowed anyway?

Reply:
Hmmm, very good point.

I don't recall Legal API doing that check, does it?

Reply:
Yes, I believe it checks if the filing is allowed.

Reply:
So, I'm trying to put things together here in my head. 

There's gonna be an external UI which will be a starting point for all these local filings. So let's say AGM extensions, it's going to route directly to for example: `https://dev.business.bcregistry.gov.bc.ca/BC0883544/agm-extension` right?

Then, I guess a check here would be something that we don't need to do. Let's leave it for the BE to take care of that like you mentioned. Thoughts Sev?

Reply:
> There's gonna be an external UI which will be a starting point for all these local filings. So let's say AGM extensions, it's going to route directly to for example: `https://dev.business.bcregistry.gov.bc.ca/BC0883544/agm-extension` right?

Yes, perhaps with a filing ID in the case of resuming a draft.

> Then, I guess a check here would be something that we don't need to do. Let's leave it for the BE to take care of that like you mentioned. Thoughts Sev?

I really liked the idea of the internal params object, as that positively prevented users from manually routing to a specific filing. I can't think of a way to do that now, unless we want to provide a signed URL query parameter or session storage variable maybe. @kialj876 , thoughts?

Reply:
I don't think you should be depending on session storage / local params because it complicates things that want to redirect here (like the new business dashboard). I don't think we need to worry about a user manually going here -- if they are a bad actor doing that they won't be able to file / there should be errors loading the page anyways. All of that needs to be handled on the backend because anyone can query the API

Reply:
Thanks Karim and Kial for thinking about this with me.

So, no change to what I've done so far. Resolving this conversation.